src/xdp/offload.c

```diff
@@ -53,7 +51,7 @@ XdpIrpInterfaceOffloadRssGetCapabilities(
         goto Exit;
     }
 
-    if ((OutputBufferLength == 0) && (Irp->Flags & IRP_INPUT_OPERATION) == 0) {
+    if ((OutputBufferLength == 0) && QueryRequiredSize) {
```

Comment:
I wonder if we can structure this internally in a way that avoids the boolean, and then do the `IRP_INPUT_OPERATION` check in the caller? That is, always return the size needed if the output buffer length is zero in this routine.

Reply:
Fixed